File: template/go/agent365_adapter.go
package main

import "fmt"

type Agent365Adapter struct {
	config AppConfig
}

func newAgent365Adapter(config AppConfig) Agent365Adapter {
	// TODO: Initialize the real Agent365 SDK client with tenant/app credentials.
	return Agent365Adapter{config: config}
}

func (a Agent365Adapter) reportTurnStart(context TurnContext) {
	// TODO: Map this to Agent365 telemetry/reporting API.
	fmt.Printf("[Agent365] Turn started: %s\n", context.TurnID)
}

func (a Agent365Adapter) reportPurviewDecision(context TurnContext, stage string, decision Decision) {
	// TODO: Map this to Agent365 telemetry/reporting API.
	fmt.Printf("[Agent365] Stage=%s, Turn=%s, Block=%t\n", stage, context.TurnID, decision.Block)
}

func (a Agent365Adapter) reportTurnEnd(context TurnContext, result ModelResult) {
	// TODO: Map this to Agent365 telemetry/reporting API.
	fmt.Printf("[Agent365] Turn ended: %s, Blocked=%t\n", context.TurnID, result.Blocked)
}

File: template/cpp/src/agent365_adapter.cpp
#include "agent365_adapter.h"

#include <iostream>

Agent365Adapter::Agent365Adapter(const AppConfig& config) : config_(config) {
  // TODO: Initialize the real Agent365 SDK client with tenant/app credentials.
}

void Agent365Adapter::reportTurnStart(const Context& context) const {
  // TODO: Map this to Agent365 telemetry/reporting API.
  std::cout << "[Agent365] Turn started: " << context.turnId << "\n";
}

void Agent365Adapter::reportPurviewDecision(
    const Context& context,
    const std::string& stage,
    const Decision& decision) const {
  // TODO: Map this to Agent365 telemetry/reporting API.
  std::cout << "[Agent365] Stage=" << stage << ", Turn=" << context.turnId
            << ", Block=" << (decision.block ? "true" : "false") << "\n";
}

void Agent365Adapter::reportTurnEnd(const Context& context, const ModelResult& result) const {
  // TODO: Map this to Agent365 telemetry/reporting API.
  std::cout << "[Agent365] Turn ended: " << context.turnId
            << ", Blocked=" << (result.blocked ? "true" : "false") << "\n";
}
